### ISSUE TITLE
TypeScript issue in @react-spring/shared for `Rafz`

### ISSUE THREAD

**Issue by sudkumar** [pmndrs/react-spring#1559]
🙏  Thank you for the latest release of 9. Love it ❤️  .
---

I am experiencing an issue when using it in TypeScript. Following are the details of the issue and possible fixes.

### Description

|Package|Version|
|:-|-:|
|@react-spring/web|9.2.2|
|typescript|4.0.2|

I am using `@react-spring/web` in one of my packages. When exporting type definitions, I got following error:

`Cannot find module 'packages/rafz' or its corresponding type declarations.`

Here is how I am generating type declaration for my package.

```sh
$ tsc src/*.tsx --outDir types --emitDeclarationOnly --declaration true --jsx preserve --target es5 --allowSyntheticDefaultImports true
.
../../node_modules/@react-spring/shared/dist/declarations/src/globals.d.ts(3,22): error TS2307: Cannot find module 'packages/rafz'
or its corresponding type declarations.
.
```

After inspection, I found that type definitions of `Rafz` were imported from `packages/rafz` instead of `@react-spring/rafz`

https://github.com/pmndrs/react-spring/blob/81ba96ce89fcec1391136a66e2f0de25ae0eb1c3/packages/shared/src/globals.ts#L1-L7

### Possible Solution

This is a small import issue which can be easily fixed by importing type definitions of `Rafz` from `@react-spring/rafz`.

```diff
- import { raf } from '@react-spring/rafz' 
+ import { raf, Rafz } from '@react-spring/rafz' 
.
.
- import { Rafz } from 'packages/rafz' 
```

 I tested it locally by updating the declaration files in my `node_modules` and it is working

----

I would love to contribute this fix if this issue seems like a bug, or 🙏  please guide for possible fixes if this is not a but.

**Comment by sudkumar** [comment]
@joeporpeglia Thanks. 🙏

> why the TS check didn't capture it

This may be a compiler configuration issue but not 100% sure. 

---

Should I create a PR or anybody else has picked it up?

**Comment by joshuaellis** [comment]
No one else will have picked it up, one man team at the moment. Go right ahead!

**Comment by sudkumar** [comment]
On it.

**Comment by sudkumar** [comment]
@joeporpeglia Created the PR #1560 . Please have a look.

**Comment by sudkumar** [comment]
Fixed in [9.2.3.](https://github.com/pmndrs/react-spring/releases/tag/v9.2.3) 

Tested and It is working. 

Thank you so much.

**Comment by joshuaellis** [comment]
Thank you for the fix! ⭐

**Comment by zposten** [comment]
@sudkumar @joshuaellis It seems like v9.2.3 isn't actually pulling in this typing fix

```
"@react-spring/web@^9.2.3":
  version "9.2.3"
  resolved "https://registry.yarnpkg.com/@react-spring/web/-/web-9.2.3.tgz#b9037719b489863ca2ad39caaacff6d315bbaea8"
  integrity sha512-dWRcgVDbO2UI9I03n/HVmCx9tY++Na+RwRzkzXv3E53BcFsjvnWGArnpj+xE/XgXiaII3ep2RmUj5jyYoukqGg==
  dependencies:
    "@react-spring/animated" "~9.2.0"
    "@react-spring/core" "~9.2.0"
    "@react-spring/shared" "~9.2.0"
    "@react-spring/types" "~9.2.0"
```

I have attempted to fix this in PR #1561
